### ISSUE TITLE
Unable to install deepspeech

### ISSUE THREAD

**Issue by yashthigale** [mozilla/DeepSpeech#3771]
Actually, I am new to deepspeech. When I am installing deepspeech package inside my Python virtual environment or inside my AWS Sagemaker environment. It gives me the following error :

ERROR: Could not find a version that satisfies the requirement deepspeech (from versions: none)
ERROR: No matching distribution found for deepspeech

**Comment by Gresliebear** [comment]
Same here I tried install deepspeech in docker deployment on google cloud but I am sure if the python version matters etc

Status: Downloaded newer image for python:3.11-slim
 ---> be2470db10f7
Step 2/10 : ENV PYTHONUNBUFFERED True
 ---> Running in a280f0c6f72c
Removing intermediate container a280f0c6f72c
 ---> 67f6a22d9077
Step 3/10 : ENV APP_HOME /back-end
 ---> Running in dac66b099f39
Removing intermediate container dac66b099f39
 ---> 40ce01092bbb
Step 4/10 : WORKDIR $APP_HOME
 ---> Running in 280bd9da0927
Removing intermediate container 280bd9da0927
 ---> 36a519dc5411
Step 5/10 : COPY . ./
Step 6/10 : RUN pip3 install --verbose deepspeech
Using pip 23.1.2 from /usr/local/lib/python3.11/site-packages/pip (python 3.11)
ERROR: Could not find a version that satisfies the requirement deepspeech (from versions: none)
ERROR: No matching distribution found for deepspeech
The command '/bin/sh -c pip3 install --verbose deepspeech' returned a non-zero code: 1
ERROR
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: step exited with non-zero status: 1

**Comment by vikasthange** [comment]
Same error I am also getting.

**Comment by Flimm** [comment]
I also get this error on Ubuntu 23.04:

```console
$ pip install deepspeech
ERROR: Could not find a version that satisfies the requirement deepspeech (from versions: none)
ERROR: No matching distribution found for deepspeech
```

The Python version is 3.11.4

**Comment by lissyx** [comment]
Project is unmaintained and thus no wheel built and updated for newer versions of Python.

**Comment by lissyx** [comment]
Marking the project archived is not my call.

**Comment by lissyx** [comment]
> > Project is unmaintained and thus no wheel built and updated for newer versions of Python.
> 
> @lissyx can we contribute? or update to newer versions of python or branch off for update? the projects claims open-source?

Of course, we moved away from TaskCluster to GitHub Actions for that purpose back in 2021, but nobody showed interest in getting more into the project while we were available to help and mentor. Given the long time, it's possible a lot of broken, but if you come up with fixes I will obviously review.

**Comment by Gresliebear** [comment]
> > > Project is unmaintained and thus no wheel built and updated for newer versions of Python.
> > 
> > 
> > @lissyx can we contribute? or update to newer versions of python or branch off for update? the projects claims open-source?
> 
> Of course, we moved away from TaskCluster to GitHub Actions for that purpose back in 2021, but nobody showed interest in getting more into the project while we were available to help and mentor. Given the long time, it's possible a lot of broken, but if you come up with fixes I will obviously review.

Awesome so for this issue would I need to upgrade to a version of python that deepspeech doesn't run on and update the code that is breaking? to fix this issue? then Merge a PR?

**Comment by lissyx** [comment]
You'd need to get back CI working again because it is likely broken since this time. Then adding support for new versions of python like e.g., https://github.com/mozilla/DeepSpeech/blob/aa1d28530d531d0d92289bf5f11a49fe516fdc86/.github/workflows/build-and-test.yml#L313 (but there are many other places to update)

**Comment by Gresliebear** [comment]
> You'd need to get back CI working again because it is likely broken since this time. Then adding support for new versions of python like e.g.,
> 
> https://github.com/mozilla/DeepSpeech/blob/aa1d28530d531d0d92289bf5f11a49fe516fdc86/.github/workflows/build-and-test.yml#L313
> 
> (but there are many other places to update)

this the yaml file for the CI? what commands do you run to test this? 

Sorry to ask just assume that I am new and a idiot right now

**Comment by lissyx** [comment]
Unfortunately, you need to know how to hack GitHub Actions to dive into this yaml file, and the build process involves building tensorflow,combined with the limitations from GitHub Actions, it ends up being non trivial. If you'd like to hack around, I'd suggest to learn GitHub Actions first.

**Comment by MarekChrappa** [comment]
> You'd need to get back CI working again because it is likely broken since this time. Then adding support for new versions of python like e.g.,
> 
> https://github.com/mozilla/DeepSpeech/blob/aa1d28530d531d0d92289bf5f11a49fe516fdc86/.github/workflows/build-and-test.yml#L313
> 
> (but there are many other places to update)

Does this mean that I shoud be able to pip install deepspeech if my python version is between  versions 3,6 and 3,9? Becouse I tried to install it with version 3,7 and it still was not possible
